libsolidity/parsing/Parser.cpp

```diff
@@ -160,17 +160,19 @@ ASTPointer<SourceUnit> Parser::parse(CharStream& _charStream)
 void Parser::parsePragmaVersion(SourceLocation const& _location, vector<Token> const& _tokens, vector<string> const& _literals)
 {
 	SemVerMatchExpressionParser parser(_tokens, _literals);
-	auto matchExpression = parser.parse();
-	if (!matchExpression.has_value())
+	variant<SemVerError, SemVerMatchExpression> matchExpressionOrError = parser.parse();
+	if (SemVerError const* matchExpressionError = get_if<SemVerError>(&matchExpressionOrError))
+	{
 		m_errorReporter.fatalParserError(
 			1684_error,
 			_location,
-			"Found version pragma, but failed to parse it. "
-			"Please ensure there is a trailing semicolon."
+			matchExpressionError->what()
```

Comment:
Since all these errors have the same error code, I think they should have a common prefix. For example:
```suggestion
			"Invalid version pragma. " + matchExpressionError->what()
```